andela/hc-gnext
# #153114593 Send alerts to the user when a job/check has failed until they respond

#### What does this PR do?
This PR adds functionality for a user to be _**nagged**_ via email until they respond to a failing check.

#### Description of Task to be completed?
- Modify `static/checks.js`, add the intervalSlider variable. This is used to form and set a slider allowing a user to choose from the range of values.
- Modify `static/my_checks.css` so as to uniquely style the interval slider.
- Modify `my_checks_desktop.html` in the templates folder to implement the UI for the slider. 
- Add fields `nag_status`, `nag_after` and `interval` to the file  `hc/api/models.py`
- Add the `interval` field to `TimeoutForm` class in `hc/front/forms.py`
- set `check.interval` to receive input from the `nag interval slider` in file _`hc/front/views.py`(line 167)_

#### How should this be manually tested?
- This task has been deployed on Heroku and can be accessed by clicking [this link here](https://nag-gnext.herokuapp.com)
- At the bottom of the screen, signup to receive alerts. A login link will be sent to your email.
- Add a check, giving it a name, timeout, grace and interval periods. By default, a created check has no name, and has a timeout of 1 hour
- A user can now be able to receive alerts on any of their created checks 

#### Any background context you want to provide?
- Currently, the system sends an alert via email to the user once a job/check goes 'up' and also once the same check goes 'down'.
- This means that for a critical system, If a check goes down and the user happens not to see it, there is a high chance of them not knowing that their system is not operating normally.
- By sending the user multiple alerts, the user can keep informed about the running status of their system. For every alert, a user is not only informed of the job that failed but also gets a summary on the status of other jobs belonging to them.
- The user can now also configure a time interval after which they should be constantly alerted once their job/check goes down. By default, this interval is set to 1 day however this can be changed to accommodate longer/shorter periods.
- Once the Job/check is back up, the alerts are no longer sent until the check goes down again.

#### What are the relevant pivotal tracker stories?
[#153114593](https://www.pivotaltracker.com/story/show/153114593)
